jenkins/ansible/roles/createNightlyBuildJob/files/utils.py

```diff
@@ -0,0 +1,143 @@
+            404: "doesn't exists"
+    }
+
+    def __init__(self, repo, copr_url=None):
+        """
+            :param unicode repo: COPR repo to work on
+            :param unicode copr_url: COPR URL to access end points
+        """
+
+        self.repo = repo
+        self.copr_url = copr_url or "https://copr.fedorainfracloud.org/"
+
+    @property
+    def api_url(self):
+        """
+            Url to API endpoint
+        """
+        return "{0}/api_2".format(self.copr_url)
+
+    def sendGetRequest(self, url, params):
+        """
+            :param unicode url: request URL
+            :param json params: params to embed in the GET request
+        """
+        # Send GET request
+        response = requests.get(url = url, params = params)
+
+        # Extract data in JSON format
+        return response.json()
+
+    def getProjectID(self, name=None, group='pki'):
+        """
+            :param unicode searchQuery: search parameter similar to search box in COPR site
+            :param unicode name: repository name to search for
+            :param unicode group: group name of the project
+        """
+        # Construct URL for project endpoint
+        url = "{0}/projects".format(self.api_url)
+
+        # Define params to be sent to the End Point
+        if name is not None and group is not None:
+            PARAMS = {'group': group, 'name': name}
+        else:
+            return None
+
+        receivedData = self.sendGetRequest(url=url, params=PARAMS)
+
+        # Extract project ID
+        for projectData in receivedData['projects']:
+            project = projectData['project']
+            if project['name'] == self.repo:
+                return project['id']
+
+        return None
+
+    def getDeleteBuildIDs(self, projectID=None, package=None, days=None):
+        """
+            :param integer projectID: project ID
+            :param unicode package: package name 
+            :param integer days: no of latest days to filter out
+        """
+        # Construct URL for builds End point
+        url = "{0}/builds".format(self.api_url)
+
+        if projectID is None or package is None or days is None:
+            raise ValueError('projectID, package and days are required') 
+        
+        PARAMS = {'project_id': projectID}
+
+        # Send GET request
+        receivedData = self.sendGetRequest(url=url, params=PARAMS) 
+        
+
+        # Extract build IDs
+        buildIDs = []
+
+        # Get current time
+        currTimeStamp = datetime.datetime.now()
+
+        for buildData in receivedData['builds']:
+            build = buildData['build']
+            timeStamp = datetime.datetime.fromtimestamp(build['submitted_on'])
+            if(timeStamp < currTimeStamp - datetime.timedelta(days=days) and
+                build['package_name'] == package):
+                buildIDs.append(build['id'])
+
+        return buildIDs
+
+    def deleteBuild(self, buildID=None, login=None, token=None):
```

Comment:
Assuming the 'client' parameter was added in the constructor, it's no longer necessary to provide the login/token parameter here.